### PR TITLE
chore: common functions for slice operations 

### PR DESCRIPTION
**Pull request by vladopajic**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3494)

### CODE REVIEW

**Review by notanatol — APPROVED** [review]

**Comment by istae** [comment]
I would ask for one more review before we merge this, so 3 in total

File: pkg/bzz/address.go [hunk 1]
```diff
@@ -105,7 +105,19 @@ func generateSignData(underlay, overlay []byte, networkID uint64) []byte {
...
 func (a *Address) Equal(b *Address) bool {
-	return a.Overlay.Equal(b.Overlay) && a.Underlay.Equal(b.Underlay) && bytes.Equal(a.Signature, b.Signature) && bytes.Equal(a.Transaction, b.Transaction)
+	if a == nil || b == nil {
+		return a == b
+	}
+
+	return a.Overlay.Equal(b.Overlay) && multiaddrEqual(a.Underlay, b.Underlay) && bytes.Equal(a.Signature, b.Signature) && bytes.Equal(a.Transaction, b.Transaction)
+}
+
+func multiaddrEqual(a, b ma.Multiaddr) bool {
```
Comment (mrekucci) [inline pkg/bzz/address.go:115]: Shouldn't that go into the utilities as well?

File: pkg/swarm/utilities.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package swarm
+
+func AddressSliceContains(addrs []Address, a Address) bool {
+	return FindAddressIdx(addrs, a) != -1
+}
+
+func AddressSliceRemove(addrs []Address, a Address) []Address {
```
Comment (mrekucci) [inline pkg/swarm/utilities.go:11]: I'd recommend renaming it to `RemoveAddress`.

File: pkg/swarm/utilities.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	i := FindAddressIdx(addrs, a)
+	if i == -1 {
+		return addrs
+	}
+
+	return append(addrs[:i], addrs[i+1:]...)
+}
+
+func FindAddressIdx(addrs []Address, a Address) int {
```
Comment (mrekucci) [inline pkg/swarm/utilities.go:20]: I'd recommend renaming it to `IndexOfAddress`.

File: pkg/swarm/utilities.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	for i, v := range addrs {
+		if v.Equal(a) {
+			return i
+		}
+	}
+	return -1
+}
+
+func FindChunkIdxWithAddress(chunks []Chunk, a Address) int {
```
Comment (mrekucci) [inline pkg/swarm/utilities.go:29]: I'd recommend renaming it to `IndexOfChunkAddress`.

File: pkg/swarm/utilities.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	for i, c := range chunks {
+		if c != nil && a.Equal(c.Address()) {
+			return i
+		}
+	}
+	return -1
+}
+
+func ChunksSliceContainsAddress(chunks []Chunk, a Address) bool {
```
Comment (mrekucci) [inline pkg/swarm/utilities.go:38]: I'd recommend renaming it to `ContainsChunkAddress`.

**Review by mrekucci — APPROVED** [review]

File: pkg/bzz/address.go (hunk 1 above)
Comment (vladopajic) [reply pkg/bzz/address.go:115]: `Multiaddr` it's not part of `bee` package.

File: pkg/swarm/utilities.go (hunk 5 above)
Comment (vladopajic) [reply pkg/swarm/utilities.go:38]: Following naming convention set with method `ContainsAddress` it implies that 
`ContainsChunkAddress` would search `[]ChunkAddress` for `ChunkAddress`.

Would `ContainsChunkWithAddress` be more suitable?

File: pkg/swarm/utilities.go (hunk 4 above)
Comment (vladopajic) [reply pkg/swarm/utilities.go:29]: Similarly to `ContainsChunkAddress`, would it be more suitable to name it `IndexOfChunkWithAddress`

**Review by mrekucci — APPROVED** [review]

**Pull request merged**